crates/vm/src/system/memory/online.rs

```diff
@@ -186,12 +190,22 @@ impl<F: PrimeField32> TracingMemory<F> {
         );
     }
 
-    fn execute_splits<const ALIGN: usize>(
+    pub(crate) fn execute_splits<const ALIGN: usize, const UPDATE_META: bool>(
```

Comment:
as mentioned, your life may be easier if `ALIGN` doesn't need to be const generic anymore